src/main/java/net/glowstone/GlowWorld.java

```diff
@@ -1809,13 +1826,25 @@ private void pulseTickMap() {
     public void requestPulse(GlowBlock block, long tickRate) {
         Location target = block.getLocation();
 
-
         if (tickRate > 0)
             tickMap.put(target, tickRate);
         else if (tickMap.containsKey(target))
             tickMap.remove(target);
     }
 
+    /**
+     * Calling this method will request that the block is ticked after the specified
+     * tix delay
+     *
+     * @param block The block to tick.
+     * @param tickRate The tick delay after which to pulse the block.
+     */
+    public void requestSinglePulse(GlowBlock block, long tickDelay) {
+        Location target = block.getLocation();
+
+        singleTickMap.put(target, worldAge + tickDelay);
```

Comment:
Should I change tickDelay to 0 if it's less than 0? I think if you want anything less than zero to be changed treated as zero it wouldn't really have an effect because it will just be run next tick and it is stored in the hashmap as `worldAge + tickDelay`. If you want I will change it though.